derive_builder/Cargo.toml

```diff
@@ -25,6 +25,7 @@ proc-macro = true
 logging = [ "log", "env_logger", "derive_builder_core/logging" ]
 struct_default = []
 skeptic_tests = ["skeptic"]
+try_setter = []
```

Comment:
The user should not need to set a feature flag to use generate try-setters.

Instead we only need some feature flag to activate the integration tests. The situation with having multiple crates in a workspace and running tests on travis is a bit complicated. When you look at [`travis.yml:57`](https://github.com/colin-kiegel/rust-derive-builder/blob/e44928318ea7f47f8f7483ce335fbd6c853024f5/.travis.yml#L57), you'll see that tests are started in the `derive_builder_test` folder with the arguments `--all --no-fail-fast --features "$CARGO_FEATURES"`. My experience is that the features are **only** passed to the `derive_builder_test` crate, but not `derive_builder` or `derive_builder_core` unless `derive_builder_test/Cargo.toml` specifies to _propagate_ the features (which is does not right now). So the easiest solution is to place all integration tests that require nightly into `derive_builder_test/tests` and use a feature flag in the `derive_builder_test` crate. That also has the advantage that the public `derive_builder` does not expose any feature flag that's useless for users.

You can use the [`compiletests`](https://github.com/colin-kiegel/rust-derive-builder/blob/e44928318ea7f47f8f7483ce335fbd6c853024f5/.travis.yml#L12-L13) feature flag and rename it to `nightlytests`. Please also change all occurences of `feature = "compiletests"`, but don't change filenames `compiletests.rs` or other occurences of the word. ;-)

Let me know if you need any further advice.